debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -238,6 +243,13 @@ public final void start(Map<String, String> props) {
         try {
             setTaskState(State.INITIAL);
             config = Configuration.from(props);
+            this.beanRegistry = new DefaultBeanRegistry();
+            this.serviceRegistry = new DefaultServiceRegistry(config, beanRegistry);
+            serviceRegistry.registerServiceProvider(new PostProcessorRegistryServiceProvider());
+            serviceRegistry.registerServiceProvider(new SnapshotLockProvider());
+            serviceRegistry.registerServiceProvider(new SnapshotQueryProvider());
+            serviceRegistry.registerServiceProvider(new SnapshotterServiceProvider());
```

Comment:
I certainly like this approach better than my original choice in the `CommonConnectorConfig` class. 

One suggestion here may be to add a protected method where you pass in the `ServiceRegistry` as an argument, and that would be where you move the `registerServiceProvider` calls. Then in the future if a specific connector has the need to register some custom service providers, the hook already exists as an overridable point. wdyt?

Reply:
Totally Agree, I was thinking the same!